tests/parametric/test_otel_span_methods.py

```diff
@@ -286,3 +286,774 @@ def test_otel_get_span_context(self, test_agent, test_library):
+                span.set_attributes({"http.request.method": "GET"})
+                span.set_attributes({"service.name": "new.service.name"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, OtelSpan(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "http.server.request"
+        assert root_span["resource"] == "otel_span_name"
+        assert root_span["service.name"] == "new.service.name"
+
+    @missing_feature(context.library == "go", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @irrelevant(context.library == "cpp", reason="library does not implement OpenTelemetry")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    def test_otel_span_span_type_override(self, test_agent, test_library):
+        """
+            Tests that the span type will be set to the contents of `span.type` when:
+            - span.type is present with some value
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_SERVER) as span:
+                span.set_attributes({"http.request.method": "GET"})
+                span.set_attributes({"span.type": "new.span.type"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, OtelSpan(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "http.server.request"
+        assert root_span["resource"] == "otel_span_name"
+        assert root_span["type"] == "new.span.type"
+
+    @missing_feature(context.library == "go", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @irrelevant(context.library == "cpp", reason="library does not implement OpenTelemetry")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    def test_otel_span_analytics_sample_rate_override(self, test_agent, test_library):
+        """
+            Tests that the metric for the analytics sample rate (_dd1.sr.eausr) will be set when:
+            - analytics.event is present with "true" or "false"
```

Comment:
TIL.
Should we test for the `false` case as well?

Reply:
I've added a new test for the `false` case in https://github.com/DataDog/system-tests/pull/1655/commits/54e8c19b437c926ec78afe79d86d35546316cf21

I don't know if there is a way of parameterizing tests for these instead of me just copy/pasting the test, I looked around but couldn't find examples of that so unsure if it is a thing.